### DISCUSSION TITLE
Prisma Register Error

### DISCUSSION THREAD

**Discussion by ifeanyidotdev** [RobertCraigie/prisma-client-py#500] [General]
i was trying to build a simple rest api with prisma-client-py and flask i have the configuration right but, i keep getting a register error from prisma-client-py

here is the code 
<img width="751" alt="Screenshot 2022-09-30 at 12 10 21 PM" src="https://user-images.githubusercontent.com/40169444/193257687-74f25ca8-d826-4543-86af-2d02832ed0bf.png">

here is the error output
<img width="850" alt="Screenshot 2022-09-30 at 12 09 52 PM" src="https://user-images.githubusercontent.com/40169444/193257757-cb031aac-3dab-4cba-acbe-0911d566c35d.png">

**Comment by RobertCraigie** [comment]
@Mremmalex I can't see in your screenshot where you're calling the `main()` function from, are you calling it anywhere?

It's also worth noting that async flask support is not tested. Because of Flask's internal architecture, their async support is implemented in a suboptimal way which breaks behaviour IIRC.

Have you looked through this Flask example? It may help you get set up: https://github.com/RobertCraigie/prisma-client-py/tree/main/examples/flask-url-shortener

**Comment by ifeanyidotdev** [comment]
<img width="688" alt="Screenshot 2022-09-30 at 2 06 23 PM" src="https://user-images.githubusercontent.com/40169444/193276200-bc1b0957-6ce4-4ff3-9356-63aab09b9541.png">


here

**Comment by RobertCraigie** [comment]
@Mremmalex `app.run()` is blocking, the line after it will not run until the server is stopped.
